includes/wikia/parser/templatetypes/handlers/DataTables.class.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+class DataTables {
+	const DATA_PORTABLE_ATTRIBUTE = 'data-portable';
+
+	/**
+	 * Mark wikitext tables coming from templates, so we can distinguish them on tables parse step
+	 *
+	 * @param $wikitext
+	 *
+	 * @return mixed
+	 */
+	public static function markTranscludedTables( $wikitext ) {
+		//check for tables
+		if ( preg_match_all( "/\\{\\|(.*)/\n", $wikitext, $matches ) ) {
+			foreach ( $matches[ 1 ] as $key => $match ) {
+				$wikitext = str_replace( $matches[ 0 ][ $key ], static::markTable( $match ), $wikitext );
+			}
+		}
+
+		return $wikitext;
+	}
+
+	public static function markDataTables( $html ) {
+		wfProfileIn( __METHOD__ );
+
+		$document = new DOMDocument();
+		$document->loadHTML( $html );
+
+		$tables = $document->getElementsByTagName( 'table' );
+		if ( $tables->length > 0 ) {
+			$xpath = new DOMXPath( $document );
+			/** @var DOMElement $table */
+			foreach ( $tables as $table ) {
+				if ( !$table->hasAttribute( static::DATA_PORTABLE_ATTRIBUTE ) &&
+					 $xpath->query( '*//*[@rowspan]|//*[@colspan]', $table )->length == 0
+				) {
+					$table->setAttribute( 'data-portable', 'true' );
+				}
+			}
+			// strip html and body tags
+			preg_match( '/<body>(.*)<\\/body>/sU', $document->saveHTML(), $match );
```

Comment:
comment is not reflecting what happens here, you're removing only <body> tag, not all of them